File: backend/src/test/java/com/NBE_4_5_2/Team5/global/config/BaseTestConfig.java
package com.NBE_4_5_2.Team5.global.config;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.context.annotation.Import;
import org.springframework.test.context.ActiveProfiles;

import jakarta.transaction.Transactional;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@ActiveProfiles("test")
@Import(TestConfig.class)
@Transactional
public @interface BaseTestConfig {
}

File: backend/src/test/java/com/NBE_4_5_2/Team5/global/config/BaseTestConfig.kt
package com.NBE_4_5_2.Team5.global.config

import jakarta.transaction.Transactional
import org.springframework.context.annotation.Import
import org.springframework.test.context.ActiveProfiles

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@ActiveProfiles("test")
@Import(TestConfig::class)
@Transactional
annotation class BaseTestConfig 

